tools/run_tests/helper_scripts/clean_xds_interop_resources.py

```diff
@@ -0,0 +1,168 @@
+# Copyright 2021 The gRPC Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import logging
+import json
+import sys
+import re
+import os
+import functools
+import subprocess
+import datetime
+from dataclasses import dataclass
+from typing import List, Any
+
+# Parses commandline arguments
+parser = argparse.ArgumentParser()
+parser.add_argument('--dry_run', action='store_true')
+args = parser.parse_args()
+
+# Type alias
+Json = Any
+
+# Configures this script
+KEEP_PERIOD = datetime.timedelta(hours=24)
+GCLOUD = os.environ.get('GCLOUD', 'gcloud')
+GCLOUD_CMD_TIMEOUT_S = datetime.timedelta(seconds=120).total_seconds()
+ZONE = 'us-central1-a'
+SECONDARY_ZONE = 'us-west1-b'
+PROJECT = 'grpc-testing'
+
+
+@functools.lru_cache()
+def get_expire_timestamp():
+    return (datetime.datetime.now() - KEEP_PERIOD).isoformat()
+
+
+def exec_gcloud(*cmds: List[str]) -> Json:
+    cmds = [GCLOUD, '--project', PROJECT] + list(cmds)
+    if 'list' in cmds:
+        # Add arguments to shape the list output
+        cmds.extend([
+            '--format', 'json', '--filter',
+            f'creationTimestamp<={get_expire_timestamp()}'
+        ])
+    if args.dry_run and 'delete' in cmds:
+        # Skip deletion for dry-runs
+        logging.debug('> Skipped: %s', " ".join(cmds))
+        return None
+    # Executing the gcloud command
+    logging.debug('Executing: %s', " ".join(cmds))
+    proc = subprocess.Popen(cmds,
+                            stdout=subprocess.PIPE,
+                            stderr=subprocess.PIPE,
+                            text=True)
+    returncode = proc.wait(timeout=GCLOUD_CMD_TIMEOUT_S)
```

Comment:
Since you used `subprocess.PIPE`, [this call could deadlock](https://docs.python.org/3/library/subprocess.html#subprocess.Popen.wait) if a long error is printed. Consider using a temporary file instead.

Reply:
Updated https://docs.python.org/3/library/subprocess.html#subprocess.Popen.communicate.

I somehow can't get `tempfile` work, `Popen` refuses to write to a temporary file. It could be a certain problem with macOS, but using `Popen.communicate()` is simpler.